src/Microsoft.Extensions.ML/ModelReloadToken.cs

```diff
@@ -44,6 +44,12 @@ public ModelReloadToken()
         /// <summary>
         /// Used to trigger the change token when a reload occurs.
         /// </summary>
-        public void OnReload() => _cts.Cancel();
+        public void OnReload()
+        {
+            if (!_cts.IsCancellationRequested)
```

Comment:
📝 The changes in this file should be reverted. It duplicates a cancellation check that appears here:

https://github.com/dotnet/runtime/blob/b2073bd41e7ba0606000e5950e3d960d1d2eb36d/src/libraries/System.Private.CoreLib/src/System/Threading/CancellationTokenSource.cs#L582